orchestration/dagster_orchestration/hca_manage/dataset.py

```diff
@@ -40,14 +47,71 @@ def run(arguments: Optional[list[str]] = None) -> None:
 
 
 def remove_dataset(args: argparse.Namespace, host: str) -> JobId:
-    hca = HcaManage(environment=args.env, data_repo_client=get_api_client(host=host))
+    hca = DatasetManager(environment=args.env, data_repo_client=get_api_client(host=host))
     return hca.delete_dataset(dataset_name=args.dataset_name, dataset_id=args.dataset_id)
 
 
 def create_dataset(args: argparse.Namespace, host: str) -> JobId:
-    hca = HcaManage(environment=args.env, data_repo_client=get_api_client(host=host))
+    hca = DatasetManager(environment=args.env, data_repo_client=get_api_client(host=host))
     return hca.create_dataset(
         dataset_name=args.dataset_name,
         billing_profile_id=args.billing_profile_id,
         schema_path=args.schema_path
     )
+
+
+@dataclass
+class DatasetManager:
```

Comment:
if i can suggest an alternative pattern here, consider approaching this less as a set of service objects and more as a set of data objects bundled with operations you can perform on those objects. for this one, it'd be less `DatasetManager('prod').delete_dataset('name', ...)` and instead `Dataset('prod', 'name', ...).delete()`

Reply:
I'm not sure how I feel about implementing this change; the `create` method has 3 parameters that the `delete` method doesn't have, and the `delete` method considers one of the `create` method parameters to be optional. To my eyes, it makes it less clear 😕

Reply:
yeah, giving this a second run-through in my head, it'd be possible to rework things to be more data-focused (with the object constructor just taking required parameters and the various methods taking the additional parameters you mentioned), but it'd require a radical enough reworking that i don't think it'll be beneficial at this point.

if we kept going down this route, certain types of function ('create a new X') would be a more natural fit to live elsewhere (e.g. 'create a snapshot' being a function under the dataset object rather than the snapshot object, 'create a dataset' being one of the base-level client operations or something), which i think would resolve some of the parameter disjunction that you're running into here, but the effort involved in planning out that refactor (and accounting for things like the dynamic dataset ID lookup in the current "delete dataset" operation) would be larger than the benefit we'd get from it, i think, and it's something we can do piecemeal later on if this code starts becoming a problem again.